### ISSUE TITLE
Issue with using tensors as training data

### ISSUE THREAD

**Issue by Abhishaike** [keras-team/keras#9271]
I've been experimenting with a neural model, and I've been forced to use tensors for my training data (due to my using of Tensorflow functions). However, there seems to be an issue where the accuracy never changes; as in, it never goes up and it never goes down. I've attached a code snippet below that reproduces the problem, does anybody know why this happens?

```python
import keras
import numpy as np
from keras.datasets import cifar10
from keras.layers import Flatten, Input, BatchNormalization, Dense, Dropout, SeparableConv2D
from keras.models import Model


(x_train, y_train), (x_test, y_test) = cifar10.load_data()
num_classes = len(np.unique(y_train))
x_train = x_train.astype('float32') / 255
y_train = np_utils.to_categorical(y_train, num_classes)

InputImg = Input(shape=(32, 32, 3))
InputFilters = SeparableConv2D(1, (3, 3), padding='same', activation='relu', kernel_regularizer='l2')(InputImg)
DenseModule = Flatten()(InputFilters)
DenseModule = Dense(128, activation='relu')(DenseModule)
DenseModule = BatchNormalization()(DenseModule)
DenseModule = Dropout(.5)(DenseModule)
DenseModule = Dense(10, activation='softmax')(DenseModule)

opt = keras.optimizers.SGD(lr=5e-3, momentum=0.9, nesterov=True)
model_all = Model(inputs=[InputImg], outputs=[DenseModule])
model_all.compile(optimizer=opt,
                  loss={(DenseModule.name).split('/')[0]:'categorical_crossentropy'},
                  metrics=['accuracy'],
                  target_tensors = [y_train])

model_all.fit(x=x_train, y=None, epochs=50, batch_size=1)
```

I should add that I've already tried 
```python
y_train = tf.convert_to_tensor(np_utils.to_categorical(y_train, num_classes), dtype = 'float32')
``` 
and the problem persisted.
